homeassistant/components/downloader/__init__.py

```diff
@@ -163,7 +193,12 @@ def do_download():
                 if final_path and os.path.isfile(final_path):
                     os.remove(final_path)
 
-        threading.Thread(target=do_download).start()
+        is_async = service.data.get(ATTR_ASYNC)
+
+        if is_async:
+            threading.Thread(target=do_download).start()
```

Comment:
Perhaps better to use an executor job for this instead of randomly creating a new thread.